### DISCUSSION TITLE
Publish many messages to multiple topics from app-services

### DISCUSSION THREAD

**Discussion by Thinhhoagn0211** [edgexfoundry/community#200] [Q&A] [answered]
Hello everyone, i am writing a app-service to sending messages which got from redis to my broker by mqtt export. However, i have read document in Edgex Foundry but i don't see anywhere supports its problem. In configuration.toml, i have tried change topic to many topics but it not works, please help me
![Screenshot from 2023-11-07 10-21-49](https://github.com/edgexfoundry/community/assets/96675317/ceb0c1d6-de17-4395-a1d7-eab477933fde)

**Comment by lenny-goodell** [comment, marked as answer]
@Thinhhoagn0211 , Nope, the built in MQTT export function from the SDK doesn't support publishing to multiple topics. I looks like the config you posted is modified from the cloud-export-mqtt example, correct? The `Topics` field should be `Topic` and only supports a single topic to export to.

If you are already creating a custom app service to do your data transform prior to export, you could also create your own custom MQTT Export function (based from the source from the SDK version) and expand it to support multiple topics.

You could also submit a feature request [Issue on the app sdk](https://github.com/edgexfoundry/app-functions-sdk-go/issues) to support export to multiple topics. We would then consider it for the Odessa 3.2 release in Spring 2024.
